### PR TITLE
vanillajames-explore-page-polishing

### PR DESCRIPTION
**Pull request by vanilla-james**
BEOBLE-1523 Make Buttons imperative, not declarative
BEOBLE-1514 Fix Settings button color
BEOBLE-1527 clicking on 'Leave chatroom' opens 'Profile edit' panel
BEOBLE-1528 Chatroom info bug - broken picture && i should not be able to remove someone in a dm chat etc.

### CODE REVIEW

File: packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions/RecentWeb3TransactionItem.tsx [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
 import UserListItem from '../../../../User/UserListItem';
+import Link from 'next/link';
+import type {
+  IUser,
+  IUserWeb3RecentTransaction,
+  IUserWeb3RecentTransactionUser,
+} from '@beoble/js-sdk';
+import { internationalFormattedNumer } from '../../../../../../helper/numbers';
 
 export type RecentWeb3TransactionItemType = {
```
Comment (vanilla-james) [inline packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions/RecentWeb3TransactionItem.tsx:17]: delete this type after deleting mock data

File: packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatGroupRecommendationsList.tsx [hunk 2]
```diff
@@ -9,22 +9,22 @@ const StyledChatGroupRecommendations = styled.ul`
   gap: 24px;
 `;
 
-const mockList = [
+export const mockChatroomsList = [
```
Comment (vanilla-james) [inline packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatGroupRecommendationsList.tsx:12]: delete mock data

File: packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions.tsx [hunk 3]
```diff
@@ -53,27 +53,43 @@ const mockRecentWeb3Transactions: RecentWeb3TransactionItemType[] = [
     // to_address: '0xeB721591677E76Ea222aC9dBa481DF5cEF7BF622', // james
     to_address: '0x1c4c1e2Af22Fa3517D41A5779Cadb66D62043da5', // james unregistered metamask
     token_address: '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619', // polygon wrapped ether
+
```
Comment (vanilla-james) [inline packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions.tsx:56]: delete mock data

File: packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/LensFriendsProfiles.tsx [hunk 4]
```diff
@@ -1,51 +1,66 @@
...
-  const { lensFriendProfiles, setLensFriendProfiles } =
-    useLensFriendProfilesAtom();
-  setLensFriendProfiles(mockLensProfiles);
-  if (!lensFriendProfiles || lensFriendProfiles?.length < 1) {
-    return null;
-  }
+  const { setLensFollowerProfiles, hasLensFriendProfiles } =
+    useLensFollowerProfilesAtom();
+  console.log('hasLensFriendProfiles', hasLensFriendProfiles);
+  // NOTE: some test handles
```
Comment (vanilla-james) [inline packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/LensFriendsProfiles.tsx:36]: delete test data

File: packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatroomSearch.tsx [hunk 5]
```diff
@@ -87,19 +89,9 @@ export default function ChatroomSearch() {
...
-      {
-        id: 'eba987a9-9c44-45f9-85aa-7e58748033a4',
-      },
-      {
-        id: 'e7dfd3fd-7847-44e9-b79c-104fcf7d73c1',
-      },
-    ]);
+    debug('onSubmit getValues', getValues());
+
+    setMockList(mockChatroomsList);
```
Comment (vanilla-james) [inline packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatroomSearch.tsx:94]: hookup to backend search api

File: packages/beoble/components/Profile/ProfileBioAndSocial.tsx [hunk 6]
```diff
@@ -127,6 +127,7 @@ const ProfileBioAndSocial = ({ userId }: { userId: string }) => {
   const { setEvent } = useFirebase();
 
   const queryClient = useQueryClient();
+  // NOTE: refactor out into re-usable hook
```
Comment (vanilla-james) [inline packages/beoble/components/Profile/ProfileBioAndSocial.tsx:130]: todo
> packages/beoble/hooks/follow/useToggleFollow.ts

File: packages/react/src/components/UserListItem/index.tsx [hunk 7]
```diff
@@ -140,6 +140,7 @@ export const UserListItem = ({
     useGetUserWeb3SpamFilterValidation();
   const { processErrorOrInfoToast } = useProcessErrorOrInfoToast();
   const usePostChatroomMutation = usePostChatroom();
+  // NOTE: creating chatroom logic to refactor out
```
Comment (vanilla-james) [inline packages/react/src/components/UserListItem/index.tsx:143]: todo
> packages/beoble/hooks/chatroom/useCreateChatroom.ts

File: packages/js-sdk/src/core/encryption.ts [hunk 8]
```diff
@@ -73,10 +73,19 @@ export class Encrypter {
   public decrypt(encryptedMessage: string | Buffer) {
     if (typeof encryptedMessage === 'string')
       encryptedMessage = Buffer.from(encryptedMessage, 'hex');
-    return decrypt(
-      this.injectedAccount.privateKey,
-      encryptedMessage
-    ).toString();
+    // NOTE: temp try catch until fix arrives
```
Comment (vanilla-james) [inline packages/js-sdk/src/core/encryption.ts:76]: todo
restore original logic

Comment (vanilla-james) [inline packages/js-sdk/src/core/encryption.ts:76]: todo
restore original logic

**Review by bamnenim-kryptonium — APPROVED** [review]

File: packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions.tsx (hunk 3 above)
Comment (vanilla-james) [reply packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions.tsx]: done

File: packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatGroupRecommendationsList.tsx (hunk 2 above)
Comment (vanilla-james) [reply packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatGroupRecommendationsList.tsx:12]: done

File: packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatroomSearch.tsx (hunk 5 above)
Comment (vanilla-james) [reply packages/beoble/components/DesktopView/Explore/Tabs/Web3CommunityChat/ChatroomSearch.tsx:94]: done

File: packages/react/src/components/UserListItem/index.tsx (hunk 7 above)
Comment (vanilla-james) [reply packages/react/src/components/UserListItem/index.tsx:143]: skipping as file in is react package
might run into circular dependencies

File: packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions/RecentWeb3TransactionItem.tsx (hunk 1 above)
Comment (vanilla-james) [reply packages/beoble/components/DesktopView/Explore/Tabs/ForYouTab/RecentWeb3Transactions/RecentWeb3TransactionItem.tsx]: keeping: these types needed regardless of testing

**Pull request merged**